### PR TITLE
[SwiftMailer] Update future reference

### PR DESCRIPTION
**Pull request by mykiwi**
- [x] Require symfony/swiftmailer-bundle#159

### CODE REVIEW

File: reference/configuration/swiftmailer.rst [hunk 1]
```diff
@@ -191,6 +225,13 @@ logging
 If true, Symfony's data collector will be activated for Swift Mailer and
 the information will be available in the profiler.
 
+.. tip::
+
+    The following configuration can used environment variable:
```
Comment (nicolas-grekas) [inline reference/configuration/swiftmailer.rst]: The following configuration options are compatible with `%env()%` parameters (with link to doc about them?)

File: reference/configuration/swiftmailer.rst [hunk 2]
```diff
@@ -37,6 +41,15 @@ Configuration
 * `disable_delivery`_
 * `logging`_
 
+url
+~~~
+
+**type**: ``string``
+
+The URL to configure swift mailer.
```
Comment (xabbuh) [inline reference/configuration/swiftmailer.rst]: Swiftmailer

Comment (xabbuh) [inline reference/configuration/swiftmailer.rst]: "The URL" doesn't look precise enough to me. What about "The URL describing the mail server to connect with"?

**Review by javiereguiluz — APPROVED** [review]
👍 

@mykiwi thanks a lot for this! Your contribution was great, but I've proposed some minor tweaks on top of it. Thanks!

File: reference/configuration/swiftmailer.rst [hunk 3]
```diff
@@ -37,6 +41,15 @@ Configuration
 * `disable_delivery`_
 * `logging`_
 
+url
+~~~
+
```
Comment (HeahDude) [inline reference/configuration/swiftmailer.rst]: Missing version added? (same for other options below)

File: reference/configuration/swiftmailer.rst [hunk 4]
```diff
@@ -37,6 +41,15 @@ Configuration
...
+**type**: ``string``
+
+The entire SwiftMailer configuration using a DSN-like URL format.
```
Comment (HeahDude) [inline reference/configuration/swiftmailer.rst]: "using" => "uses"

Comment (javiereguiluz) [inline reference/configuration/swiftmailer.rst]: I think "using" is correct here. We're not saying that Swift "uses" this ... but that the "url" option is all the Swift config using a DSN format.

File: reference/configuration/swiftmailer.rst [hunk 5]
```diff
@@ -97,20 +100,29 @@ timeout
 
 **type**: ``integer``
 
+.. versionadded:: 2.4.2
+    The ``url`` option was added in SwiftMailerBundle 2.4.2.
```
Comment (HeahDude) [inline reference/configuration/swiftmailer.rst]: The parameter looks wrong here and below.

Comment (javiereguiluz) [inline reference/configuration/swiftmailer.rst]: 😭  it's fixed now!

File: reference/configuration/swiftmailer.rst [hunk 6]
```diff
@@ -191,6 +237,14 @@ logging
 If true, Symfony's data collector will be activated for Swift Mailer and
 the information will be available in the profiler.
 
+.. tip::
+
+    The following configuration options are compatible with ``%env()%`` parameters:
+    ``url``, ``transport``, ``username``, ``password``, ``host``,
+    ``port``, ``timeout``, ``source_ip``, ``local_domain``,
+    ``encryption``, ``auth_mode``.
```
Comment (HeahDude) [inline reference/configuration/swiftmailer.rst]: Considering https://github.com/symfony/swiftmailer-bundle/pull/159#issuecomment-283193936, we should remove these two parameters for now, wdyt?

**Review by HeahDude — APPROVED** [review]
Thanks @mykiwi @javiereguiluz

**Comment by javiereguiluz** [comment]
Merging it because the related PR was already merged. Thanks Romain!

**Pull request merged**